kornia/filters/otsu_thresholding.py

```diff
@@ -0,0 +1,285 @@
+# LICENSE HEADER MANAGED BY add-license-header
+#
+# Copyright 2018 Kornia Team
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import annotations
+
+from typing import Optional, Tuple, Union
+
+import torch
+
+from kornia.core.check import KORNIA_CHECK
+from kornia.utils.image import perform_keep_shape_image
+
+
+class ThreshOtsu(torch.nn.Module):
+    """Otsu thresholding module for PyTorch tensors."""
+
+    def __init__(self, nbins: int = 256) -> None:
+        """Initialize the ThreshOtsu module.
+
+        Args:
+            nbins (int, optional): Number of bins for histogram computation. Default is 256.
+
+        Attributes:
+            nbins (int): Number of bins for histogram computation.
+            _threshold (Union[float, torch.Tensor]): Otsu-computed threshold, initialized to -1.
+            _early_stop (float): Fraction of max iterations without improvement before early stopping.
+        """
+        super().__init__()
+        self.nbins: int = nbins
+
+        self._threshold: Union[float, torch.Tensor] = -1
```

Comment:
why can't these be passed as args as well?

Reply:
Yes nbins can surely be moved to args instead of class attribute. I will make the change. For _threshold, I can return it along with the result thresholded (as done by cv2)